### PR TITLE
fix EnumAdapter crash when running under PIX

### PR DESCRIPTION
**Pull request by jseward**
When running under PIX (Debug->Graphics->Start Graphics Debugging) the call to factory6->EnumAdapterByGpuPreference will fail so gracefully fallback to EnumAdapters1.

NOTE: This code fix should probably be applied all copies of the cut-and-paste code in the samples.

### CODE REVIEW

**Comment by walbourn** [comment]
The fix here is actually less code-churn...

Change ``DXGI_ERROR_NOT_FOUND !=`` tests to ``SUCCEEDED(`` macro test instead.

**Comment by jseward** [comment]
Ha fair enough, I didn't realize `D3D12CreateDevice` accepted a null adapter and basically does this fallback anyways.

**Comment by sebmerry** [comment]
@jseward thank you for your contribution!  If you would like to make the changes recommended by Chuck (and minimizing the other code/comment changes), I'd be happy to complete this PR and replicate the change in the other samples.

File: Samples/Desktop/D3D12HelloWorld/src/HelloTriangle/DXSample.cpp [hunk 1]
```diff
@@ -49,59 +49,48 @@ void DXSample::GetHardwareAdapter(
 
     ComPtr<IDXGIAdapter1> adapter;
 
-    ComPtr<IDXGIFactory6> factory6;
-    if (SUCCEEDED(pFactory->QueryInterface(IID_PPV_ARGS(&factory6))))
-    {
-        for (
-            UINT adapterIndex = 0;
-            DXGI_ERROR_NOT_FOUND != factory6->EnumAdapterByGpuPreference(
```
Comment (sebmerry) [inline Samples/Desktop/D3D12HelloWorld/src/HelloTriangle/DXSample.cpp:57]: As per Chuck: Just change this (and others below) to SUCCEEDED()

File: Samples/Desktop/D3D12HelloWorld/src/HelloTriangle/DXSample.cpp [hunk 2]
```diff
@@ -49,59 +49,48 @@ void DXSample::GetHardwareAdapter(
...
-
-            // Check to see whether the adapter supports Direct3D 12, but don't create the
-            // actual device yet.
-            if (SUCCEEDED(D3D12CreateDevice(adapter.Get(), D3D_FEATURE_LEVEL_11_0, _uuidof(ID3D12Device), nullptr)))
-            {
-                break;
-            }
-        }
-    }
-    else
```
Comment (sebmerry) [inline Samples/Desktop/D3D12HelloWorld/src/HelloTriangle/DXSample.cpp:81]: I suspect the other change you will need is to make this an if(adapter.Get() == nullptr) check.

**Review by sebmerry — CHANGES_REQUESTED** [review]

**Comment by jseward** [comment]
Hi @sebmerry, honestly the proper fix would involve refactoring out `DXSample.cpp` to be shared by all samples so the fix doesn't have to be copy/pasted to 62 different files. Probably best for your team to look into. Feel free to close this PR, reference for the underlying issue that needs to be fixed.

**Comment by sebmerry** [comment]
Sounds good, I'll get a fix out this week. Thanks again!

**Pull request closed without merging**